### PR TITLE
Adding Pathway Information Into Platemaps

### PR DESCRIPTION
**Pull request by axiomcura**
This PR introduces the updated platemaps that include pathway information for the compounds. To organize the files better, the following directory changes were made:
- The original platemaps have been moved to a new directory named `original_platemaps`.
- The augmented platemaps, which now include pathway information, are stored in the `updated_platemaps` directory.

I am open to suggestions on better naming conventions for these directories to ensure they clearly reflect their purpose and contents.

Additionally, I added a new Jupyter notebook, `process_metadata.ipynb`, located in the `metadata` folder. This notebook processes the original platemaps, integrates the pathway information, and generates the updated platemaps. The corresponding Python script, converted from the notebook, has been added to the `scripts` folder.

### CODE REVIEW

File: utils/io_utils.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+"""
+This module provides functions for loading, writing, and modifying files in this data 
+repository. It centralizes input/output operations, ensuring efficient and consistent 
+handling of data files throughout the project.
+"""
+import pathlib
+from collections import defaultdict
+import pandas as pd
+
+def load_barcodes(barcode_path: str | pathlib.Path) -> dict:
```
Comment (gwaybio) [inline utils/io_utils.py:10]: Looks like this `.py` file needs a quick autoformatting tool applied? (e.g., ruff)? (there should be two new line characters before function definitions, I believe)

File: metadata/scripts/process_metadata.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        plate_pathway_meta_df,
+        left_on=["treatment", "well_position"],
+        right_on=["UCD ID", "Well"],
+        how="left",
+    )
+
+    # Ensure no rows or columns in the original platemap were modified during the merge
+    # This checks that only the added MOA information changes the dataframe,
+    # while the original content remains identical.
+    assert merged_df[
```
Comment (gwaybio) [inline metadata/scripts/process_metadata.py:81]: Great to see this!

File: metadata/process_metadata.ipynb [hunk 3]
```diff
@@ -0,0 +1,229 @@
+{
```
Comment (gwaybio) [inline metadata/process_metadata.ipynb:1]: the name of this file could be improved. What type of metadata are you processing? What do you mean by "process"?

Perhaps `metadata/update_platemaps_and_barcodes_to_add_pathway_info.ipynb` is a better (albeit long) name?

**Review by gwaybio — APPROVED** [review]
Looks great @axiomcura - a super easy PR to review. I have a few comments which probably should be resolved prior to merging.

When you make file name changes, please remember to remove the older file. A `git mv FILENAME NEW_FILENAME` command would do this trick.

File: metadata/process_metadata.ipynb (hunk 3 above)
Comment (axiomcura) [reply metadata/process_metadata.ipynb]: I updated it to update_platemap_and_barcodes_with_pathway_info.ipynb.

**Comment by axiomcura** [comment]
@gwaybio I have applied all the changes, unfortunately I am not able to merge, I do not have write access.

**Comment by gwaybio** [comment]
Thanks for letting me know - I just added you as a maintainer

**Comment by axiomcura** [comment]
merged! thank you!

**Pull request merged**